From d116939663e9c6eafbd21698e286652715c73946 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Wed, 22 May 2019 01:37:19 -0700
Subject: [PATCH 1150/1200] mmc: Don't print error when probe returns -ENODEV

It is possible for the probe to return -ENODEV or -ENXIO
if the mmc device is not present.  Do not print an error
message in this case.  This matches the Linux kernel
behavior.

Change-Id: I1c527207872f8ce5b8dc79aba54153c7dea89855
Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/9718
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
---
 drivers/mmc/mmc.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index 868b8843a3..c8f0869c93 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -2825,8 +2825,14 @@ static int mmc_probe(bd_t *bis)
 	}
 	uclass_foreach_dev(dev, uc) {
 		ret = device_probe(dev);
-		if (ret)
-			pr_err("%s - probe failed: %d\n", dev->name, ret);
+		if (ret) {
+			if (ret == -ENODEV || ret == -ENXIO)
+				pr_debug("%s: probe of %s rejects match %d\n",
+					 __func__, dev->name, ret);
+			else
+				pr_err("%s - probe failed: %d\n",
+				       dev->name, ret);
+		}
 	}
 
 	return 0;
-- 
2.22.0

